src/dune_rules/pkg_rules.ml

```diff
@@ -1214,16 +1517,13 @@ end = struct
         resolve_impl
     in
     fun (db : DB.t) ctx (loc, name) ->
-      if Package.Name.Set.mem db.system_provided name
-      then Memo.return `System_provided
-      else
-        Memo.exec memo (db, ctx, name)
-        >>| function
-        | Some s -> `Inside_lock_dir s
-        | None ->
-          User_error.raise
-            ~loc
-            [ Pp.textf "Unknown package %S" (Package.Name.to_string name) ]
+      Memo.exec memo (db, ctx, name)
```

Comment:
The old behavior did not pointlessly memoize the look up in `db.system_rpovided`. It doesn't really matter, but I'd like to understand why this was changed.